pkg/util/ipvs/ipvs_linux.go

```diff
@@ -0,0 +1,459 @@
+}
+
+func (runner *runner) setSystemFlagInt(sysControl string, value int) error {
+	if val, err := runner.sysctl.GetSysctl(sysControl); err == nil && val != value {
+		runner.sysctl.SetSysctl(sysControl, value)
+	} else if err != nil {
+		glog.Errorf("Error: System control flag [%s] cannot be set", sysControl)
+		return err
+	}
+	return nil
+}
+
+func (runner *runner) CheckAliasDevice(aliasDev string) error {
+	_, err := net.InterfaceByName(aliasDev)
+	if err == nil {
+		return nil
+	}
+	err = runner.CreateAliasDevice(aliasDev)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getAllAlias(aliasDev string) ([]net.IP, error) {
+	i, err := net.InterfaceByName(aliasDev)
+	if err != nil {
+		return nil, err
+	}
+	addrs, err := i.Addrs()
+	if err != nil {
+		return nil, err
+	}
+	ips := make([]net.IP, 0)
+	for _, addr := range addrs {
+		ip, _, err := net.ParseCIDR(addr.String())
+		if err != nil {
+			return nil, err
+		}
+		ips = append(ips, ip)
+	}
+	return ips, nil
+
+}
+
+func (runner *runner) CreateAliasDevice(aliasDev string) error {
+
+	if aliasDev == AliasDevice {
+		// Generate device alias
+		args := []string{"link", "add", aliasDev, "type", "dummy"}
+		if _, err := runner.exec.Command(cmd, args...).CombinedOutput(); err != nil {
+			// "exit status 2" is returned from the above run command if the device already exists
+			if !strings.Contains(fmt.Sprintf("%v", err), "exit status 2") {
+				glog.Errorf("Error: Cannot create alias network device: %s", aliasDev)
+				return err
+			}
+			glog.V(6).Infof(" Info: Alias network device already exists and skip create: args: %s", args)
+			return nil
+		}
+		glog.V(6).Infof(" Succeeded: Create alias device: %s", aliasDev)
+	}
+
+	return nil
+
+}
+
+func (runner *runner) DeleteAliasDevice(aliasDev string) error {
+	if aliasDev == AliasDevice {
+		// Delete device alias
+		args := []string{"link", "del", aliasDev}
+		if _, err := runner.exec.Command(cmd, args...).CombinedOutput(); err != nil {
+			// "exit status 1" is returned from the above run command if the device don't exists
+			if !strings.Contains(fmt.Sprintf("%v", err), "exit status 1") {
+				glog.Errorf("Error: Cannot delete alias network device: %s", aliasDev)
+				return err
+			}
+			glog.V(6).Infof(" Info: Alias network device don't exists and skip delete: args: %s", args)
+			return nil
+		}
+		glog.V(6).Infof(" Succeeded: Delete alias device: %s", aliasDev)
+	}
+
+	return nil
+}
+
+func (runner *runner) SetAlias(serv *Service) error {
+	// TODO:  Hard code command to config aliases to network device
+	ips, err := getAllAlias(AliasDevice)
+	var found bool = false
+	if err == nil {
+		for _, ip := range ips {
+			if ip.Equal(serv.Address) {
+				found = true
+				break
+			}
+		}
+	}
+	if found {
+		return nil
+	}
+	// Generate device alias
+	alias := AliasDevice + ":" + strconv.FormatUint(uint64(IPtoInt(serv.Address)), 10)
+	args := []string{"addr", "add", serv.Address.String(), "dev", AliasDevice, "label", alias}
```

Comment:
Thanks. I will try the way you suggested. Before doing that, would you please tell me what's the advantage of adding a route to the `local` table when compared to creating a dummy interface?